File: InterfaceExample.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package interfaceexample;

/**
 *
 * @author Aaron
 */

public class InterfaceExample {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
            IntPenguin MyPenguin = new IntPenguin();

            IntDuck MyDuck = new IntDuck();

            System.out.println("Check first method:");
            MyPenguin.swim();
            MyDuck.swim();

            System.out.println("Check second method:");
            MyPenguin.walk();
            MyDuck.walk();

            System.out.println("The penguin is " + MyPenguin.age + " years old");
            System.out.println("The duck is " + MyDuck.age + " years old");
    }
}

File: InterfaceExample.cs
using System;

namespace InterfaceExample
{

    public interface NonFlyingBird {
        void walk();
        void swim();

        int Age {
            get;
            set;
        }
    }

    public class Penguin : NonFlyingBird {
        private int penguinAge = 1;
        public int Age {
            get{
                return penguinAge;
            }
            set {
                penguinAge = value;
            }
        }
        public void walk() {
            Console.WriteLine("The penguin waddles");
        }

        public void swim() {
            Console.WriteLine("The penguin swims like a champion");
        }
    }

    public class Duck : NonFlyingBird {
        private int duckAge = 2;
        public int Age {
            get{
                return duckAge;
            }
            set {
                duckAge = value;
            }
        }
        public void walk() {
            Console.WriteLine("The duck walks like a normal bird");
        }

        public void swim() {
            Console.WriteLine("The duck swims, but not as well as the penguin");
        }
    }
    class Program
    {
        static void Main(string[] args)
        {
            Penguin MyPenguin = new Penguin();

            Duck MyDuck = new Duck();

            Console.WriteLine("Check first method:");
            MyPenguin.swim();
            MyDuck.swim();

            Console.WriteLine("Check second method:");
            MyPenguin.walk();
            MyDuck.walk();

            Console.WriteLine("The penguin is " + MyPenguin.Age + " years old");
            Console.WriteLine("The duck is " + MyDuck.Age + " years old");

        }
    }
}
